arcgis-ios-sdk-samples/Edit data/Add features (feature service)/AddFeaturesViewController.swift

```diff
@@ -16,81 +16,94 @@ import UIKit
+    func addFeature(at mapPoint: AGSPoint) {
+        // Disable interaction with map view.
         mapView.isUserInteractionEnabled = false
         
-        // normalize geometry
-        let normalizedGeometry = AGSGeometryEngine.normalizeCentralMeridian(of: mappoint)!
+        // Normalize geometry.
+        let normalizedGeometry = AGSGeometryEngine.normalizeCentralMeridian(of: mapPoint)!
         
-        // attributes for the new feature
+        // Attributes for the new feature.
         let featureAttributes = ["typdamage": "Minor", "primcause": "Earthquake"]
-        // create a new feature
+        // Create a new feature.
         let feature = featureTable.createFeature(attributes: featureAttributes, geometry: normalizedGeometry)
         
-        // show the progress hud
-        UIApplication.shared.showProgressHUD(message: "Adding..")
-        
-        // add the feature to the feature table
+        // Add the feature to the feature table.
         featureTable.add(feature) { [weak self] (error: Error?) in
-            UIApplication.shared.hideProgressHUD()
+            guard let self = self else { return }
+            // Enable interaction with map view.
+            self.mapView.isUserInteractionEnabled = true
             
             if let error = error {
-                self?.presentAlert(message: "Error while adding feature: \(error.localizedDescription)")
+                self.presentAlert(message: "Error while adding feature: \(error.localizedDescription)")
             } else {
-                // applied edits on success
-                self?.applyEdits()
+                // Applied edits on success.
+                self.applyEdits()
             }
-            // enable interaction with map view
-            self?.mapView.isUserInteractionEnabled = true
         }
     }
     
+    /// Apply local edits to the geodatabase.
     func applyEdits() {
-        self.featureTable.applyEdits { [weak self] (featureEditResults: [AGSFeatureEditResult]?, error: Error?) in
-            if let error = error {
-                self?.presentAlert(message: "Error while applying edits :: \(error.localizedDescription)")
-            } else {
-                if let featureEditResults = featureEditResults,
-                    featureEditResults.first?.completedWithErrors == false {
-                    self?.presentAlert(message: "Edits applied successfully")
-                }
-                UIApplication.shared.hideProgressHUD()
+        guard serviceGeodatabase.hasLocalEdits() else { return }
+        serviceGeodatabase.applyEdits { [weak self] featureTableEditResults, error in
+            if let featureTableEditResults = featureTableEditResults,
+               featureTableEditResults.first?.editResults.first?.completedWithErrors == false {
+            } else if let error = error {
+                self?.presentAlert(message: "Error while applying edits: \(error.localizedDescription)")
             }
```

Comment:
It seems odd to have an empty clause. Is it enough to check the error?

```suggestion
        serviceGeodatabase.applyEdits { [weak self] _, error in
            if let error = error {
                self?.presentAlert(message: "Error while applying edits: \(error.localizedDescription)")
            }
```

Reply:
Oof, I thought linter could catch this sort of mistake 🤦